### ISSUE TITLE
zero_snr DDIM causing black image:

### ISSUE THREAD

**Issue by jloganolson** [huggingface/diffusers#5103]
I am also having this issue - to reproduce the black image: 
```
pipe = StableDiffusionXLPipeline.from_pretrained(SDXL_PATH)
pipe.scheduler = DDIMScheduler.from_config(
    pipe.scheduler.config, rescale_betas_zero_snr=True, timestep_spacing="trailing"
)
pipe.to("cuda")

prompt = "A lion in galaxies, spirals, nebulae, stars, smoke, iridescent, intricate detail, octane render, 8k"
pipe(prompt, guidance_rescale=0.7).images[0]
```
You also get this error: `diffusers/image_processor.py:65): RuntimeWarning: invalid value encountered in cast
  images = (images * 255).round().astype("uint8")`

If you use the basic DDIM scheduler, the image is generated `pipe.scheduler = DDIMScheduler.from_config(pipe.scheduler.config)`

float16 and safetensors don't appear to affect it so left that out in reproduction but you can use this to reproduce as well:
`pipe = StableDiffusionXLPipeline.from_pretrained(SDXL_PATH,torch_dtype=torch.float16, variant="fp16", use_safetensors=True)`

_Originally posted by @jloganolson in https://github.com/huggingface/diffusers/issues/4195#issuecomment-1726070750_

**Comment by jloganolson** [comment]
update it looks like the latents are going to -inf/inf/nan on scheduler.step - looking into it more...

**Comment by jloganolson** [comment]
Update: You can get rid of the inf/nan by changing the scheduler config 'prediction_type' to v_scheduler (`scheduler_config['prediction_type']="v_prediction"`) but it just outputs grey (no error tho)

**Comment by patrickvonplaten** [comment]
Does it also give black images in full fp32 precision?

**Comment by jloganolson** [comment]
Per the top line there, isn't that the default? Or do I need to set something explicitly? Repost:
```
pipe = StableDiffusionXLPipeline.from_pretrained(SDXL_PATH)
pipe.scheduler = DDIMScheduler.from_config(
    pipe.scheduler.config, rescale_betas_zero_snr=True, timestep_spacing="trailing"
)
pipe.to("cuda")
```
If so, yes this gets black and the error message

**Comment by patrickvonplaten** [comment]
I can reproduce the issue. Note that when changing `rescale_betas_zero_snr=True` to `rescale_betas_zero_snr=False` the issue seems to be solved:

```py
import hf_image_uploader as hiu
from diffusers import DDIMScheduler
from diffusers import AutoPipelineForText2Image
import torch

pipe = AutoPipelineForText2Image.from_pretrained("stabilityai/stable-diffusion-xl-base-1.0", variant="fp16", torch_dtype=torch.float16)
pipe.scheduler = DDIMScheduler.from_config(
    pipe.scheduler.config, rescale_betas_zero_snr=False, timestep_spacing="trailing"
)
pipe.to("cuda")

prompt = "A lion in galaxies, spirals, nebulae, stars, smoke, iridescent, intricate detail, octane render, 8k"
image = pipe(prompt, guidance_rescale=0.7).images[0]

hiu.upload(image, repo_id="patrickvonplaten/images")
```

**Comment by patrickvonplaten** [comment]
In order for `rescale_betas_zero_snr=True` to work one has to problably fine-tune SDXL on the v-prediction objective as only this way the model can tackle 0 final SNR (also cc @bghira here)

**Comment by bghira** [comment]
that's interesting. i believe someone on the EveryDream2 discord server had demonstrated useful results with epsilon models on zsnr but I've not replicated that experiment. that said, I have tested SDXL with the zsnr code and it should at least produce results? maybe the math really just doesn't work out. but that would be odd. the original paper authors (cc @PeterL1n ) mentioned that merely epsilon "does not meaningfully learn", but did not state that it outright broke

**Comment by AmericanPresidentJimmyCarter** [comment]
It gave me black images too, but I thought this was the intended result with eps pred.

**Comment by LinB203** [comment]
I re-trained the model with v_pred & zsnr, but if run 300 step inference step, it still return nan. In fewer step such as 100/200 step, it is ok.

**Comment by bghira** [comment]
way too many steps. just do 25
